src/lib.rs

```diff
@@ -0,0 +1,497 @@
+            .build()
+            .context("could not initiate an SDKMS client")?;
+
+        info!("create primary key");
+        let primary = PublicKey::create(
+            &http_client,
+            key_name.to_string(),
+            KeyRole::Primary,
+            algorithm,
+        )?;
+
+        info!("create decryption subkey");
+        let subkey = PublicKey::create(
+            &http_client,
+            key_name.to_string(),
+            KeyRole::Subkey,
+            algorithm,
+        )?;
+
+        let mut agent = PgpAgent {
+            api_endpoint: api_endpoint.to_string(),
+            api_key: api_key.to_string(),
+            primary,
+            subkey,
+            certificate: None,
+        };
+
+        agent
+            .bind_sdkms_keys_and_generate_cert(user_id)
+            .context("could not generate public certificate")?;
+
+        Ok(agent)
+    }
+
+    /// Signs the given content and writes the detached signature to `sink`.
+    pub fn sign_detached(
+        self,
+        sink: &mut (dyn Write + Send + Sync),
+        content: &[u8],
+        armored: bool,
+    ) -> Result<()> {
+        let sequoia_key =
+            &self.primary.sequoia_key.expect("unloaded primary key");
+
+        let signer = RawSigner {
+            api_endpoint: &self.api_endpoint,
+            api_key:      &self.api_key,
+            descriptor:   &self.primary.descriptor,
+            public:       &sequoia_key,
+        };
+
+        let mut message = Message::new(sink);
+        if armored {
+            message = Armorer::new(message).build()?;
+        }
+
+        let mut message = Signer::new(message, signer).detached().build()?;
+        message.write_all(content)?;
+        message.finalize()?;
+
+        Ok(())
+    }
+
+    /// Decrypts the given ciphertext and writes the plaintext to `sink`.
+    pub fn decrypt(
+        &mut self,
+        sink: &mut dyn Write,
+        ciphertext: &[u8],
+        policy: &(dyn PgpPolicy + 'static),
+    ) -> Result<()> {
+        struct Helper<'a> {
+            decryptor: RawDecryptor<'a>,
+        }
+
+        impl VerificationHelper for Helper<'_> {
+            fn get_certs(
+                &mut self,
+                _ids: &[KeyHandle],
+            ) -> openpgp::Result<Vec<Cert>> {
+                Ok(Vec::new())
+            }
+
+            fn check(
+                &mut self,
+                _structure: MessageStructure,
+            ) -> openpgp::Result<()> {
+                Ok(())
+            }
+        }
+
+        impl DecryptionHelper for Helper<'_> {
+            fn decrypt<D>(
+                &mut self,
+                pkesks: &[PKESK],
+                _skesks: &[SKESK],
+                sym_algo: Option<SymmetricAlgorithm>,
+                mut decrypt: D,
+            ) -> openpgp::Result<Option<openpgp::Fingerprint>>
+            where
+                D: FnMut(SymmetricAlgorithm, &SessionKey) -> bool,
+            {
+                pkesks[0]
```

Comment:
Are we sure that this element exists?

Reply:
Good catch. Normally, yes, but I'm adding a check for the sake of modularity.